js/modules/authApiUtils.js

```diff
@@ -0,0 +1,107 @@
+import { $emailInput, $passwordInput, $passwordVerifyInput } from "./domElements.js";
+import { triggerInputValidationError } from "./authDOMHandler.js";
+
+const API_URL = "https://bootcamp-api.codeit.kr/api";
+
+// accessToken 존재여부 확인 후 리다이렉트 하는 함수
+const redirectIfAccessTokenExists = (directURL) => {
+  const accessToken = localStorage.getItem("accessToken");
+  if (accessToken) {
+    location.replace(directURL);
+  }
+};
+
+// 로그인 api에 접근하는 함수
+const postSignIn = async (user) => {
+  try {
+    const response = await fetch(`${API_URL}/sign-in`, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(user),
+    });
+    // code 200이 아닐 시 에러 발생으로 동작 중지
+    if (!response.ok) {
+      throw new Error();
+    }
```

Comment:
요청 자체에는 성공했지만, ok 필드가 falsy하다면 에러를 던지겠군요. 어디서 무엇때문에 난 에러인지 메시지도 있으면 좋겠네요. 



```suggestion
    if (!response.ok) {
      throw new Error(response.message); // or ('로그인 실패')
    }
```

에러를 던지는 방식에 대해서 실무적으로는...사용자경험에 문제가 있던없던, 모니터링 툴에서 보면 수만개가 찍혀있는 경우도 있으니, 이유,맥락이 같이 있으면 좀 더 편하게 빨리 고칠 수 있겠죠